interfaces/builtin/socketcan.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const socketCanSummary = `allows configuration and use of SocketCAN network interfaces`
+
+const socketCanBaseDeclarationSlots = `
+  socketcan:
+    allow-installation:
+      slot-snap-type:
+        - core
+`
+
+const socketCanConnectedPlugAppArmor = `
+# Description: Can access SocketCAN networks.
+network can,
+network netlink raw,
+
+capability net_admin,
+
+# Allow configuration of the interface using ip
+/{,usr/}{,s}bin/ip ixr,
```

Comment:
This seems to imply there would be a canbus-control and regular canbus interface (without the ability to configure can).

Reply:
I can answer @zyga's questions. The ip command will require net_admin for setting up the interface and also needs netlink raw for some operations. Allow of this is covered by network-control, but it seems useful to not lump SocketCAN into all other networking since it is not so much about network connectivity in the traditional sense as about using the socket() API in the kernel.

As such, please adjust your rules to be:
```
# Description: Can configure and use SocketCAN networks
network can,

# Allow configuration of the interface using the ip command for SocketCAN
/{,usr/}{,s}bin/ip ixr,

# required by ip to configure SocketCAN networks
capability net_admin,
network netlink raw,
```